src/main/java/org/ossgang/commons/mapbackeds/MapbackedObject.java

```diff
@@ -0,0 +1,96 @@
+package org.ossgang.commons.mapbackeds;
+
+import static java.util.Objects.requireNonNull;
+
+import java.lang.invoke.MethodHandles;
+import java.lang.invoke.MethodType;
+import java.lang.reflect.InvocationHandler;
+import java.lang.reflect.Method;
+import java.util.Arrays;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+
+/**
+ * This is the 'Implementation' of a mapbacked object. It shall never instantiated directly by the user, but always
+ * factory methods or builders from the Mapbackeds class shall be used.
+ * The responsibilities of this class are:
+ * <ul>
+ * <li>delegate getter calls to appropriate lookups in the internal map</li>
+ * <li>delegate calls to equals(), hashCode() and toString() to the internal implementations</li>
+ * <li>delegate calls to default methods of the proxied interface to the real implementations</li>
+ * </ul>
+ */
+class MapbackedObject implements InvocationHandler {
+
+    private final Class<?> intfc;
+    private final Map<String, Object> fieldValues;
+    private final Set<Method> fieldMethods;
+
+    MapbackedObject(Class<?> intfc, Map<String, Object> fieldValues) {
+        this.intfc = requireNonNull(intfc, "interface must not be null");
+        this.fieldValues = Map.copyOf(fieldValues);
```

Comment:
This is java 11 only ;-)

Reply:
fixed (copying to new HashMap in constructor and getter.